Epmgcipgab 18 footer quick links

suren-tadevosyan opened the pull request:
Summary
Added `Footer` and `QuickLinks` components.

Details
- Implemented `Footer` component with collapsible sections and responsive design.
- Added `QuickLinks` section with background images and responsive layout.
- Styled components to ensure proper alignment and functionality on various screen sizes

suren-tadevosyan committed:
Add footer component  (incomplete)

suren-tadevosyan committed:
Add Footer and QuickLinks components

suren-tadevosyan committed:
Verified responsiveness on various screen sizes.

andreyBalabukha commented on package.json:
```diff
@@ -15,7 +15,8 @@
     "react": "^18",
-    "react-dom": "^18"
+    "react-dom": "^18",
+    "react-icons": "^5.2.1"
```
Why are you using `React-icons` library?

suren-tadevosyan replied:
I used the FaChevronDown icon from the react-icons library to indicate the CollapsibleBox component's title is collapsible.

andreyBalabukha replied:
Right, but why `React-icons` library, and not any other library? is it make sense to save icons separately from figma and don't include library?

suren-tadevosyan replied:
Got it, I will proceed with sourcing icons from Figma and integrating them into our application.

andreyBalabukha commented on src/app/[locale]/globals.css:
```diff
@@ -1,3 +1,5 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
```
please remove redundant line

suren-tadevosyan replied:
done

andreyBalabukha commented on src/components/boxes/collapsible-box/collapsible-box.tsx:
```diff
@@ -0,0 +1,47 @@
+import { FaChevronDown } from "react-icons/fa";
+import Link from "next/link";
+
+interface CollapsibleBoxProps {
```
if you are using interface, please name it with the `I....` : `ICollapsibleBoxProps `

suren-tadevosyan replied:
done

andreyBalabukha commented on src/components/footer/index.tsx:
```diff
@@ -0,0 +1,90 @@
+            className="text-lg text-center  mt-4"
+            style={{ color: "white" }}
+          >
+            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
```
Please all literals put to the separate fiel

suren-tadevosyan replied:
done

andreyBalabukha commented on src/components/footer/index.tsx:
```diff
@@ -0,0 +1,90 @@
+      <Box className="border-t border-gray-800 text-white">
+        <Section className="flex justify-center">
+          <Text className="text-sm" style={{ color: "white" }}>
+            © 2024 GABT named after A.Navoiy. All rights reserved. Designed by{" "}
```
Please all literals put to the separate file and delete redundant space

suren-tadevosyan replied:
done

andreyBalabukha commented on src/components/footer/index.tsx:
```diff
@@ -0,0 +1,90 @@
+            © 2024 GABT named after A.Navoiy. All rights reserved. Designed by{" "}
+            <Link href="https://www.epam.com" className="hover:text-dark-red">
+              <Text className="text-sm" style={{ color: "white" }}>
+                Epam
```
Please all literals put to the separate file

suren-tadevosyan replied:
done

andreyBalabukha commented on src/components/intro/annonce-section/index.tsx:
```diff
@@ -19,9 +19,9 @@ export const AnnonceSection = () => {
       </Section>
       <Section className="flex flex-row justify-center">
-        <Image className="px-8 " src={annonceImage1} alt="annonce-1" />
+        {/* <Image className="px-8 " src={annonceImage1} alt="annonce-1" />
```
Why did you comment images?

andreyBalabukha commented on src/components/intro/language-switcher/index.tsx:
```diff
@@ -0,0 +1,60 @@
+  if (!isMounted) return null;
+
+  const getLanguageName = (locale: Locale): string => {
+    return locale === "en" ? "Eng" : "Рус";
```
Please all literals put to the separate file

suren-tadevosyan replied:
done

andreyBalabukha commented on src/constants/footer-items-mock.ts:
```diff
@@ -0,0 +1,24 @@
+    { text: "Terms of Service", href: "/terms" },
+    { text: "Support", href: "/support" },
+  ];
+  
+
```
Please remove redundant lines

suren-tadevosyan replied:
done

andreyBalabukha commented on src/components/footer/index.tsx:
```diff
@@ -0,0 +1,90 @@
+import insta from "../../../public/insta.png";
+
+const Footer: React.FC = () => {
+  const [visibleBox, setVisibleBox] = useState<string | null>(null);
```
Can we use a better naming? For example `currentFooterSection` or smt like that

suren-tadevosyan replied:
done

suren-tadevosyan committed:
Implemented suggested changes

suren-tadevosyan committed:
./src/components/footer/index.tsx

andreyBalabukha commented on src/app/[locale]/globals.css:
```diff
@@ -1,3 +1,3 @@
 @tailwind base;
 @tailwind components;
-@tailwind utilities;
+@tailwind utilities;
```
please add blank line

andreyBalabukha commented on src/components/footer/index.tsx:
```diff
@@ -0,0 +1,90 @@
+import Image from "next/image";
+import CollapsibleBox from "../boxes/collapsible-box/collapsible-box";
+import {
+  BOX1_ITEMS,
```
Could you make a better naming, for example `FOOTER_SECTION_ONE`

andreyBalabukha commented on src/components/intro/language-switcher/index.tsx:
```diff
@@ -0,0 +1,60 @@
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+import { localeLiterals } from "../../../constants/local-literals";
+
+type Locale = "en" | "ru";
```
Could you please make "en" and "ru" as enums that will be imported

andreyBalabukha commented on src/components/intro/quick-links/index.tsx:
```diff
@@ -0,0 +1,78 @@
+import { Box, Section, Text } from "@radix-ui/themes";
+import Link from "next/link";
+import { Bona_Nova } from "next/font/google";
+import linkimage1 from "../../../../public/quicklinks1.png";
```
I think import `alias` like `@` should work by default.

andreyBalabukha commented on src/components/intro/quick-links/index.tsx:
```diff
@@ -0,0 +1,78 @@
+          }}
+        >
+          <Section className="w-full h-full flex justify-center items-center border">
+            <Link href="/our-team" passHref>
```
The routes also can be enums

andreyBalabukha commented on src/components/intro/quick-links/index.tsx:
```diff
@@ -0,0 +1,78 @@
+              <Text
+                className={`${bonnaNova.className} text-4xl font-semibold transition-colors duration-300`}
+              >
+                Our Team
```
please add to literals

andreyBalabukha commented on src/components/intro/quick-links/index.tsx:
```diff
@@ -0,0 +1,78 @@
+              <Text
+                className={`${bonnaNova.className} text-4xl font-semibold transition-colors duration-300`}
+              >
+                Dress Code
```
please add to literals

suren-tadevosyan committed:
Update footer component, collapsible-box, package-lock, and package.json

andreyBalabukha commented on src/components/footer/index.tsx:
```diff
@@ -41,15 +41,41 @@ const CurrentFooterSection: React.FC = () => {
+                <div className="bg-red-500 rounded-full border border-white hover:border-golden-yellow py-2 px-3 hover:bg-golden-yellow group">
+                  <Image
+                    src={facebook}
+                    alt="Instagram"
```
Is that ALT correct?

andreyBalabukha commented on src/components/footer/index.tsx:
```diff
@@ -41,15 +41,41 @@ const CurrentFooterSection: React.FC = () => {
+                <div className="bg-red-500 rounded-full border border-white hover:border-golden-yellow p-2 hover:bg-golden-yellow group">
+                  <Image
+                    src={teleg}
+                    alt="Instagram"
```
Is that ALT correct?

andreyBalabukha commented on src/components/footer/index.tsx:
```diff
@@ -72,13 +98,15 @@ const CurrentFooterSection: React.FC = () => {
-              </Text>
+            <Link href="https://www.epam.com">
+              <Image src={EpamLogo} alt="EPAM" className="ml-2 mt-1 
+              " />
```
why that code on other line?

andreyBalabukha commented on package-lock.json:
If you didn't update packages, why do you have changes in that file?

andreyBalabukha commented on public/telegram.png:
For the icons and images, if we have a choice between `PNG` and `SVG`, please use svg, because it is much more flexible. Would you mind to update to svg?

suren-tadevosyan committed:
Addressed PR comments:
- Fixed route issues and converted routes to enums.
-  Moved Quick Links literals to a separate quickLinksLiterals.ts.
- Updated the  component to use the literals and dynamically apply styles based on the new properties.

andreyBalabukha commented on src/components/intro/language-switcher/index.tsx:
```diff
@@ -0,0 +1,60 @@
+  useEffect(() => {
+    setIsMounted(true);
+    const locale = pathname.split("/")[1] as Locale;
+    if (locale === "en" || locale === "ru") {
```
I wanted you to create enums - to use in such cases. like:
`if (locale === Locale.en || locale === Locale.ru)`
make sense?

suren-tadevosyan replied:
I've implemented locale enums in the language switcher branch . However, I haven’t included these enums in the footer branch to avoid mixing changes at this stage. 

If needed, I can merge the language switcher branch into the footer branch

andreyBalabukha commented on src/components/intro/language-switcher/index.tsx:
```diff
@@ -0,0 +1,60 @@
+      </DropdownMenu.Trigger>
+
+      <DropdownMenu.Content>
+        {currentLocale !== "en" && (
```
Same here

andreyBalabukha commented on src/components/intro/language-switcher/index.tsx:
```diff
@@ -0,0 +1,60 @@
+            {localeLiterals.dropdownItems.en}
+          </DropdownMenu.Item>
+        )}
+        {currentLocale !== "ru" && (
```
And here

andreyBalabukha commented on src/components/intro/welcome-text/index.tsx:
```diff
@@ -7,15 +7,16 @@ const bonnaNova = Bona_Nova({ subsets: ["latin"], weight: "400" });
       <Section className="w-full flex flex-col justify-center items-center">
         <Heading
-          className={`${bonnaNova.className} mb-1.5 lg:text-8xl md:text-5xl sm:text-5xl text-center`}
+          className={`${bonnaNova.className} mb-1.5 lg:text-8xl md:text-5xl sm:text-5xl text-center  `}
```
Do we really need additional space here?

andreyBalabukha commented on src/components/intro/welcome-text/index.tsx:
```diff
@@ -7,15 +7,16 @@ const bonnaNova = Bona_Nova({ subsets: ["latin"], weight: "400" });
           {t("home-page-welcome")}
         </Heading>
-        <Text className="block w-2/3 lg:text-3xl md:text-2xl sm:text-2xl text-center">
+        <Text className="block w-2/3 lg:text-3xl md:text-2xl  sm:text-2xl text-center  ">
```
Do we really need additional space here?

suren-tadevosyan committed:
Merge branch 'develop' into EPMGCIPGAB-18-footer-quick-links

balabukha committed:
fix ru.json
